### PR TITLE
fixing re-following

### PR DESCRIPTION
**Pull request by 2is10**
also introducing a new convention for state filtering:
- `INACTIVE` filtered out by default
- turn off filtering by passing `excludeState = None`

### CODE REVIEW

File: shoebox/app/com/keepit/model/Follow.scala [hunk 1]
```diff
@@ -39,11 +39,10 @@ case class Follow (
 
 @ImplementedBy(classOf[FollowRepoImpl])
 trait FollowRepo extends Repo[Follow] {
-  def all(userId: Id[User])(implicit session: RSession): Seq[Follow]
-  def get(uriId: Id[NormalizedURI])(implicit session: RSession): Seq[Follow]
-  def get(userId: Id[User], uriId: Id[NormalizedURI])(implicit session: RSession): Option[Follow]
-  def getByUrlId(urlId: Id[URL])(implicit session: RSession): Seq[Follow]
-  def getByUriId(urlId: Id[NormalizedURI])(implicit session: RSession): Seq[Follow]
+  def get(userId: Id[User], uriId: Id[NormalizedURI], excludeState: Option[State[Follow]] = Some(FollowStates.INACTIVE))(implicit session: RSession): Option[Follow]
+  def getByUser(userId: Id[User], excludeState: Option[State[Follow]] = Some(FollowStates.INACTIVE))(implicit session: RSession): Seq[Follow]
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Follow.scala:43]: nice, better then just a get

File: shoebox/app/com/keepit/model/Follow.scala [hunk 2]
```diff
@@ -63,29 +62,17 @@ class FollowRepoImpl @Inject() (val db: DataBaseComponent) extends DbRepo[Follow
     def * = id.? ~ createdAt ~ updatedAt ~ userId ~ uriId ~ urlId.? ~ state <> (Follow, Follow.unapply _)
   }
 
-  def all(userId: Id[User])(implicit session: RSession): Seq[Follow] =
-    (for(f <- table if f.userId === userId && f.state === FollowStates.ACTIVE) yield f).list
+  def get(userId: Id[User], uriId: Id[NormalizedURI], excludeState: Option[State[Follow]])(implicit session: RSession): Option[Follow] =
+    (for (f <- table if f.uriId === uriId && f.userId === userId && f.state =!= excludeState.getOrElse(null)) yield f).firstOption
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Follow.scala:66]: so if the state is empty it just checks for "state is not null" ?

Comment (2is10) [inline shoebox/app/com/keepit/model/Follow.scala:66]: That's right, and `state` is defined as a `NOT NULL` column. I don't think it'll slow the database down measurably.

Comment (eishay) [inline shoebox/app/com/keepit/model/Follow.scala:66]: sure

**Pull request merged**